src/server/apis/last-viewed.ts

```diff
@@ -0,0 +1,50 @@
+import * as express from 'express';
+
+import {GenericStatusResponse} from '../../types/api';
+import {userModel} from '../models/userModel';
+
+import {createAPIRoute, ResponseDetails} from './api-route';
+
+async function handleUpdateRequest(request: express.Request):
+    Promise<ResponseDetails<GenericStatusResponse>> {
+  const userRecord = await userModel.getUserRecordFromRequest(request);
+  if (!userRecord) {
+    return {
+      statusCode: 400,
```

Comment:
Use `401 Unauthorized` instead.

Reply:
Technically this should never be reached because we check the user is authorized in the login middleware - but will change regardless.